### PR TITLE
chore: add end-to-end test for PB write service

### PR DESCRIPTION
**Pull request by jacobmarble**
Add a simple end-to-end test for the gRPC/PB write service.

- [x] I've read the contributing section of the project [README](https://github.com/influxdata/influxdb_iox/blob/main/README.md).
- [x] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed).

### CODE REVIEW

**Comment by jacobmarble** [comment]
TIL
```rust
assert!(matches!(...))
```

File: entry/src/entry.rs [hunk 1]
```diff
@@ -380,31 +379,23 @@ fn pb_check_table_batch_column_types(table_batch: &pb::TableBatch) -> Result<()>
...
         lp_field_detected,
-        time_nontime_detected,
-        time_time_detected,
+        lp_tag_detected,
+        time_column_detected,
     ) {
-        (true, false, false, _, _) => Ok(()), // Expected IOx column set
-        (false, _, true, false, true) => Ok(()), // Expected line protocol column set
-        (true, true, _, _, _) => Err(Error::PBSemanticTypeConflict {
+        (false, true, _, true) => Ok(()), // Expected line protocol column set
```
Comment (alamb) [inline entry/src/entry.rs:386]: I don't really understand these changes as they seem unrelated to the end to end test (perhaps they are issues the end to end test uncovered?)

**Review by alamb — APPROVED** [review]
Looks good to me

File: entry/src/entry.rs (hunk 1 above)
Comment (jacobmarble) [reply entry/src/entry.rs:386]: Correct. Tests revealed that the unreachable match arm was reachable.

**Review by e-dard — APPROVED** [review]
This LGTM 👍
